Default to small icons and Sans font.

diff -urp inkscape-0.45+0.46pre2.orig/src/preferences-skeleton.h inkscape-0.45+0.46pre2/src/preferences-skeleton.h
--- inkscape-0.45+0.46pre2.orig/src/preferences-skeleton.h	2008-02-17 05:24:14.000000000 +0100
+++ inkscape-0.45+0.46pre2/src/preferences-skeleton.h	2008-02-22 09:34:12.000000000 +0100
@@ -73,7 +73,7 @@ static char const preferences_skeleton[]
 "                       tracebackground=\"0\" usepressure=\"1\" usetilt=\"0\" keep_selected=\"1\"/>\n"
 "    <eventcontext id=\"text\"  usecurrent=\"0\" gradientdrag=\"1\"\n" 
 "                       font_sample=\"AaBbCcIiPpQq12369$\342\202\254\302\242?.;/()\"\n"
-"                  style=\"fill:black;fill-opacity:1;stroke:none;font-family:Bitstream Vera Sans;font-style:normal;font-weight:normal;font-size:40px;\" selcue=\"1\"/>\n"
+"                  style=\"fill:black;fill-opacity:1;stroke:none;font-family:Sans;font-style:normal;font-weight:normal;font-size:40px;-inkscape-font-specification:Sans;font-stretch:normal;font-variant:normal\" selcue=\"1\"/>\n"
 "    <eventcontext id=\"nodes\" selcue=\"1\" gradientdrag=\"1\" show_handles=\"1\" sculpting_profile=\"1\" />\n"
 "    <eventcontext id=\"tweak\" selcue=\"0\" gradientdrag=\"0\" show_handles=\"0\" width=\"0.2\" force=\"0.2\" fidelity=\"0.5\" usepressure=\"1\" style=\"fill:red;stroke:none;\" usecurrent=\"0\"/>\n"
 "    <eventcontext id=\"gradient\" selcue=\"1\"/>\n"
@@ -308,7 +308,7 @@ static char const preferences_skeleton[]
 "    <group\n"
 "       id=\"tools\"\n"
 "       icononly=\"1\"\n"
-"       small=\"0\" />\n"
+"       small=\"1\" />\n"
 "  </group>\n"
 "\n"
 "  <group\n"
