--- inkscape-0.41/src/xml/sp-repr.h.gcc4	2005-05-10 05:38:09.000000000 -0400
+++ inkscape-0.41/src/xml/sp-repr.h	2005-05-10 05:38:40.000000000 -0400
@@ -44,6 +44,7 @@
 class SPRepr : public Inkscape::GC::Anchored {
 public:
     SPRepr() {}
+    virtual ~SPRepr() {}
 
     virtual SPReprType type() const=0;
 
--- inkscape-0.41/src/file.cpp.gcc4	2005-05-10 05:40:00.000000000 -0400
+++ inkscape-0.41/src/file.cpp	2005-05-10 05:40:52.000000000 -0400
@@ -269,7 +269,7 @@
     Glib::ustring::size_type const dataLen = ustr.length();
     Glib::ustring::size_type const cstrLen = strlen(cstr);
 
-    g_message("   size: %d\n   length: %d\n   bytes: %d\n    clen: %d", ustr.size(), dataLen, byteLen, cstrLen );
+    g_message("   size: %zd\n   length: %zd\n   bytes: %zd\n    clen: %zd", ustr.size(), dataLen, byteLen, cstrLen );
     g_message( "  ASCII? %s", (ustr.is_ascii() ? "yes":"no") );
     g_message( "  UTF-8? %s", (ustr.validate() ? "yes":"no") );
 
--- inkscape-0.41/src/interface.cpp.gcc4	2005-02-01 02:11:06.000000000 -0500
+++ inkscape-0.41/src/interface.cpp	2005-05-10 05:29:12.000000000 -0400
@@ -332,7 +332,7 @@
         if (event->state & GDK_SHIFT_MASK) shortcut |= SP_SHORTCUT_SHIFT_MASK;
         if (event->state & GDK_CONTROL_MASK) shortcut |= SP_SHORTCUT_CONTROL_MASK;
         if (event->state & GDK_MOD1_MASK) shortcut |= SP_SHORTCUT_ALT_MASK;
-        sp_shortcut_set (shortcut, (Inkscape::Verb *)((int)data), true);
+        sp_shortcut_set (shortcut, (Inkscape::Verb *)data, true);
     }
 }
 
