--- src/libnrtype/Layout-TNG-OutIter.cpp~	2018-12-04 08:57:55.000000000 -0600
+++ src/libnrtype/Layout-TNG-OutIter.cpp	2019-01-11 08:22:33.026490234 -0600
@@ -26,7 +26,7 @@
     double best_x_difference = DBL_MAX;
 
     if (char_index == _characters.size()) return end();
-    for ( ; char_index < _characters.size() ; char_index++) {
+    for ( ; char_index + 1 < _characters.size() ; char_index++) {
         if (_characters[char_index].chunk(this).in_line != line_index) break;
         if (_characters[char_index].char_attributes.is_mandatory_break) break;
         if (!_characters[char_index].char_attributes.is_cursor_position) continue;
