diff -rupN inkscape-0.48.4/configure.ac inkscape-0.48.4-new/configure.ac
--- inkscape-0.48.4/configure.ac	2012-12-15 17:50:19.205918190 +0100
+++ inkscape-0.48.4-new/configure.ac	2014-05-15 00:29:24.076377009 +0200
@@ -263,7 +263,7 @@ if test "x$gc_ok" = "xyes" && test "x$cr
 			# include <gc.h>
 			#endif
 			#include <stdio.h>
-			extern unsigned GC_version;
+			unsigned GC_version = GC_get_version();
 			int main(void){
 				unsigned min = ((6 << 16) | (4 << 8) | 0);
 				printf("%d.%d.%d ",GC_version >> 16, (GC_version >> 8) & 0xFF, GC_version & 0xFF);
