--- inkscape-0.45.1/share/extensions/ill2svg.pl.orig	2007-03-25 14:40:18.000000000 +0200
+++ inkscape-0.45.1/share/extensions/ill2svg.pl	2007-03-25 14:41:47.000000000 +0200
@@ -355,20 +355,9 @@
 print " xmlns:xlink=\"http://www.w3.org/1999/xlink\">\r\n";
 
 while (<>) {
-    if (m/$NL_DOS$/) { 
-	$/ = $NL_DOS; 
-	foreach (split /$NL_DOS/) {
-	    process_line($_);
-	}
-    } elsif (m/$NL_MAC$/) {
-	$/ = $NL_MAC;
-	foreach (split /$NL_MAC/) {
-	    process_line($_);
-	}
-    } else {
-	chomp;
+    foreach (split /[\015\012]+/) {
 	process_line($_);
-    }
+      }
 }
 print "</svg>\n";
 
