diff -Nur inkscape-0.42-orig/configure inkscape-0.42/configure
--- inkscape-0.42-orig/configure	2005-07-25 07:25:35.000000000 +0200
+++ inkscape-0.42/configure	2005-07-29 12:34:14.000000000 +0200
@@ -21344,7 +21344,7 @@
 int
 main ()
 {
-return (int) ngettext ("","", 1)
+return (int)(long) ngettext ("","", 1)
   ;
   return 0;
 }
@@ -21402,7 +21402,7 @@
 int
 main ()
 {
-return (int) dgettext ("","")
+return (int)(long) dgettext ("","")
   ;
   return 0;
 }
